src/AnalyticsEventReporter.ts

```diff
@@ -1,38 +1,78 @@
-import { AnalyticsEventService } from "./AnalyticsEventService"
-import { AnalyticsConfig } from "./AnalyticsConfig";
-import { EventPayload } from "./EventPayload";
-import { EventAPIResponse } from "./EventAPIResponse";
-import { merge } from "./merge";
+import { AnalyticsEventService } from './AnalyticsEventService';
+import { AnalyticsConfig } from './AnalyticsConfig';
+import { EventPayload, PartialPayload } from './EventPayload';
+import { EventAPIResponse } from './EventAPIResponse';
+import { getOrSetupSessionId } from './setupSessionId';
+import { name, version } from '../package.json';
+import { post } from './post';
+import merge from './merge';
+import { setupRequestUrl } from './setupRequestUrl';
 
 /** Represents an reporter is responsible for reporting analytics events. */
 export class AnalyticsEventReporter implements AnalyticsEventService {
-
     private config: AnalyticsConfig;
     private payload?: EventPayload;
-
     /**
      * @param config - necessary analytics config: Must provide one and only
      * one of API Key or Bearer Token.
-     * 
+     *
      * @param payload - (optional) desired event values to report
      */
     constructor(config: AnalyticsConfig, payload?: EventPayload) {
-        const apiKeyIsSet = config.key !== undefined;
```

Comment:
Did you use prettier?

Reply:
Likely because of a linter. Undoing this indentation leads to vscode complaining about the indentation on every line. I believe we have prettier installed in the project.

Reply:
> Did you use prettier?

I realized have eslint installed in vscode by default so I think there may be some additional formatting being done on top of prettier.